### ISSUE TITLE
Rename Package MsTest.Sdk to Microsoft.NET.Sdk.Test

### ISSUE THREAD

**Issue by Varorbc** [microsoft/testfx#2693]
## Summary

https://www.nuget.org/packages/MSTest.Sdk

Microsoft provides most of the package name is the full name, I hope the package name is also the full name, so that users feel is a system, Not an alternative.Just like the following names

```
Microsoft.NET.Sdk
Microsoft.NET.Sdk.Web
Microsoft.NET.Sdk.Worker
Microsoft.NET.Sdk.BlazorWebAssembly
```

**Comment by Nirmal4G** [comment]
> comment from https://github.com/dotnet/sdk/issues/25797#issuecomment-2051597066

I thought Microsoft was good at naming. You guys disappoint me. You have loads of names at your disposal. Come on! Pick a good one that is simple and is a one word change to the standard SDK Name.

Some Examples:

- `Microsoft.Test.Sdk`
- `Microsoft.Testing.Sdk`
- `Microsoft.NET.Test.Sdk`
- `Microsoft.NET.Sdk.Test`
- `Microsoft.NET.Sdk.Testing`
- `Microsoft.NET.Testing.Sdk`

Pick one from above that seems intuitive to users who are already familiar with existing packages. Please do note that `Microsoft.NET.Test.Sdk` package can only be used as a package reference. So, why not convert to an SDK, ship in-band with the .NET SDK and deprecate the package reference functionality after v18?

**Comment by MarcoRossignoli** [comment]
@Nirmal4G thanks for the suggestion, here I've explained because at the moment use some of that names could be problematic for future evolution https://github.com/microsoft/testfx/issues/2693#issuecomment-2051698214

I think that MSBuild SDK for testing make sense if at the end the usage allows an "empty" csproj like https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk otherwise simply move 1 reference from the user project file to the sdk won't help a lot with the transparent evolution of testing frameworks.

**Comment by martincostello** [comment]
As it stands, the suggestions in this comment https://github.com/microsoft/testfx/issues/2693#issuecomment-2052075533 don't seem to make it specific that the SDK is specifically for _MSTest_. From the blog posts so far, it doesn't seem like the new test SDK is intended to support NUnit, xunit etc. as they aren't neccessarily compatible with things like native AoT.

**Comment by Evangelink** [comment]
I think it would only makes sense to reword the `MSTest.Sdk` package under one of the following conditions:

- we start to integrate MSTest SDK into .NET SDK
- we globally rename MSTest to be `Microsoft.Testing.Framework` 
- we get many upvotes

**Comment by Nirmal4G** [comment]
@Evangelink I also propose a new Shared Framework of testing tools and frameworks in .NET SDK along with a new Test SDK for the Project (*not just a simple rename as requested here). May be not for v4 but for v5 maybe? .NET 10? ***Basically, a built-in Testing support from the .NET SDK without any external dependencies.***

See #3254 for the proposal and discussion.

**Comment by Youssef1313** [comment]
I don't think we will want to rename `MSTest.Sdk`. The name is clear enough IMO, and makes it specific to MSTest. So I'm closing the issue.
